src/core/linux/SDL_udev.c

```diff
@@ -302,6 +302,48 @@ bool SDL_UDEV_GetProductInfo(const char *device_path, struct input_id *inpid, in
     return true;
 }
 
+bool SDL_UDEV_GetProductSerial(const char *device_path,const char **serial)
+{
+    struct stat statbuf;
+    char type;
+    struct udev_device *dev;
+    const char *val;
+    int class_temp;
+
+    if (!_this) {
+        return false;
+    }
+
+    if (stat(device_path, &statbuf) == -1) {
+        return false;
+    }
+
+    if (S_ISBLK(statbuf.st_mode)) {
+        type = 'b';
+    }
+    else if (S_ISCHR(statbuf.st_mode)) {
+        type = 'c';
+    }
+    else {
+        return false;
+    }
+
+    dev = _this->syms.udev_device_new_from_devnum(_this->udev, type, statbuf.st_rdev);
+
+    if (!dev) {
+        return false;
+    }
+
+    val = _this->syms.udev_device_get_property_value(dev, "ID_SERIAL_SHORT");
+    if (val) {
+        *serial = val;
+        return true;
+    }
+
+    serial = NULL;
```

Comment:
```suggestion
    *serial = NULL;
```
Actually we should do this at the top of the function so it's set before any `return false` statements. In this particular case it's not a public API and we know we're not going to use the value on a false return, so we can probably omit this.